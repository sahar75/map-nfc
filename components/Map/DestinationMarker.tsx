import React, { useMemo } from "react";
import { Text, View } from "react-native";
import { Callout, Marker } from "react-native-maps";
const locationImg = require("../../assets/images/destination.png");

interface IProps {
  selectedRoutePoints: {
    lat: number;
    lng: number;
  }[];
}

const DestinationMarker: React.FC<IProps> = ({ selectedRoutePoints }) => {
  const destination = useMemo(
    () => selectedRoutePoints?.[(selectedRoutePoints?.length ?? 0) - 1],
    [selectedRoutePoints]
  );

  return (
    <Marker
      coordinate={{
        latitude: destination?.lat ?? 0,
        longitude: destination?.lng ?? 0,
      }}
      title="Destination"
      zIndex={100}
      pinColor="#5251FA"
      image={locationImg}
    >
      <Callout
        style={{
          width: 300,
          height: 100,
        }}
      >
        <View>
          <Text>test</Text>
        </View>
      </Callout>
    </Marker>
  );
};

export default React.memo(DestinationMarker);
