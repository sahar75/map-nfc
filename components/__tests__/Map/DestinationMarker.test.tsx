import React from "react";
import renderer from "react-test-renderer";
import { Marker } from "react-native-maps";
import DestinationMarker from "../../Map/DestinationMarker";

jest.mock("react-native-maps", () => {
  return {
    Marker: jest.fn().mockReturnValue(null),
    Callout: jest.fn().mockReturnValue(null),
  };
});

describe("DestinationMarker Component", () => {
  const mockSelectedRoutePoints = [
    { lat: 37.7749, lng: -122.4194 },
    { lat: 34.0522, lng: -118.2437 },
  ];

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <DestinationMarker selectedRoutePoints={mockSelectedRoutePoints} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the Marker with correct coordinates", () => {
    const component = renderer.create(
      <DestinationMarker selectedRoutePoints={mockSelectedRoutePoints} />
    );
    const markerInstance = component.root.findByType(Marker);

    // Check if the Marker has the correct props for latitude and longitude
    expect(markerInstance.props.coordinate.latitude).toBe(34.0522);
    expect(markerInstance.props.coordinate.longitude).toBe(-118.2437);
  });

  it("uses default coordinates if no destination is available", () => {
    const component = renderer.create(
      <DestinationMarker selectedRoutePoints={[]} />
    );
    const markerInstance = component.root.findByType(Marker);

    // Check if default coordinates are used
    expect(markerInstance.props.coordinate.latitude).toBe(0);
    expect(markerInstance.props.coordinate.longitude).toBe(0);
  });
});
