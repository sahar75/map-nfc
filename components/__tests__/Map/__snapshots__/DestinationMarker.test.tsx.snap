// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DestinationMarker Component renders correctly 1`] = `null`;
